auslib/web/admin/views/permissions.py

```diff
@@ -144,23 +143,36 @@ def _delete(self, username, permission, changed_by, transaction):
 
 
 class PermissionScheduledChangesView(ScheduledChangesView):
+    """/scheduled_changes/permissions"""
+
     def __init__(self):
         super(PermissionScheduledChangesView, self).__init__("permissions", dbo.permissions)
 
     @requirelogin
     def _post(self, transaction, changed_by):
         change_type = connexion.request.json.get("change_type")
+        connexion.request.json.pop("csrf_token", None)
 
-        if change_type == "update":
-            form = ScheduledChangeExistingPermissionForm()
-        elif change_type == "insert":
-            form = ScheduledChangeNewPermissionForm()
-        elif change_type == "delete":
-            form = ScheduledChangeDeletePermissionForm()
-        else:
-            return Response(status=400, response="Invalid or missing change_type")
+        what = {}
+        for field in connexion.request.json:
+            if change_type == "delete" and field == "options" or change_type == "insert" and field == "data_version":
+                continue
+
+            what[field] = connexion.request.json[field]
 
-        return super(PermissionScheduledChangesView, self)._post(form, transaction, changed_by)
+        if what.get("options", None):
+            what["options"] = json.loads(what.get("options"))
+
+        if change_type in ["update", "delete"]:
+            if not what.get("data_version", None):
+                return problem(400, "Bad Request", "Missing field", ext={"exception": "data_version is missing"})
+            else:
+                what["data_version"] = int(what["data_version"])
+
+        elif change_type != "insert":
+            return problem(400, "Bad Request", "Invalid or missing change_type")
```

Comment:
Makes sense. Also I think we can safely remove the else check entirely since both change_type field either present in [query](https://github.com/aksareen/balrog/blob/6d5dc12245ab906bc47de15d4fa8994862f7e892/auslib/web/admin/swagger/api.yaml#L2009) or in [body](https://github.com/aksareen/balrog/blob/6d5dc12245ab906bc47de15d4fa8994862f7e892/auslib/web/admin/swagger/api.yaml#L3043).